From: Carsten Schoenert <c.schoenert@t-online.de>
Date: Sun, 29 Jun 2014 13:48:26 +0200
Subject: adopt the gridvis-server configuration

Changing the order and the value structure of etc/server.conf to
reflect the LSB folder for persistent modified date by the application
and better readability of the configuration.
---
 etc/server.conf | 76 +++++++++++++++++++++++++++++++++++++++++++++++++++------
 1 file changed, 69 insertions(+), 7 deletions(-)

diff --git a/etc/server.conf b/etc/server.conf
index 256cc32..254302d 100644
--- a/etc/server.conf
+++ b/etc/server.conf
@@ -1,19 +1,81 @@
-# ${HOME} will be replaced by user home directory according to platform
-default_userdir="/opt/GridVisData"
+# vim: set filetype=conf:
 
+################################
+# gridvis-server configuration #
+################################
 
 # options used by the launcher by default, can be overridden by explicit
 # command line switches
-default_options="--branding server -J-Dpasw-feature-energy-calc-builder=true -J-Dpasw-feature-ivalue-flat-persistence=true -J-Dpasw-feature-linechart-v2=true -J-Xms64m -J-Xmx1024m -J-Dvisualvm.display.name=GridVisService -J-Dnetbeans.mainclass=de.janitza.pasw.arg.catcher.MainClass -J-Djava.util.logging.config.class=de.janitza.pasw.pasw.logback.JulHandler -J-Dnetbeans.exception.report.min.level=99999 -J-Dnetbeans.exception.alert.min.level=99999 -J-D.level=INFO -J-Dorg.netbeans.core.startup.TopLogging.level=INFO -J-Dorg.netbeans.core.startup.NbEvents.level=INFO -J-Djava.net.preferIPv4Stack=true -J-Dorg.apache.commons.logging.Log=org.apache.commons.logging.impl.Jdk14Logger -J-Dde.janitza.pasw.device.Communication.level=WARNING -J-DserverPort=8080 --nosplash --nogui  -J-Djava.awt.headless=true -J-DSkipStoringEndBeforeStart -J-DSkipStoringTimeInFuture -J-DSkipStoringTimeTooFarInPast -J-DSkipStoringTimebaseInvalid"
+#default_options="--branding server -J-Dpasw-feature-energy-calc-builder=true -J-Dpasw-feature-ivalue-flat-persistence=true -J-Dpasw-feature-linechart-v2=true -J-Xms64m -J-Xmx1024m -J-Dvisualvm.display.name=GridVisService -J-Dnetbeans.mainclass=de.janitza.pasw.arg.catcher.MainClass -J-Djava.util.logging.config.class=de.janitza.pasw.pasw.logback.JulHandler -J-Dnetbeans.exception.report.min.level=99999 -J-Dnetbeans.exception.alert.min.level=99999 -J-D.level=INFO -J-Dorg.netbeans.core.startup.TopLogging.level=INFO -J-Dorg.netbeans.core.startup.NbEvents.level=INFO -J-Djava.net.preferIPv4Stack=true -J-Dorg.apache.commons.logging.Log=org.apache.commons.logging.impl.Jdk14Logger -J-Dde.janitza.pasw.device.Communication.level=WARNING -J-DserverPort=8080 --nosplash --nogui  -J-Djava.awt.headless=true -J-DSkipStoringEndBeforeStart -J-DSkipStoringTimeInFuture -J-DSkipStoringTimeTooFarInPast -J-DSkipStoringTimebaseInvalid"
 #-J-Dcom.sun.management.jmxremote.port=3333 -J-Dcom.sun.management.jmxremote.authenticate=false -J-Dcom.sun.management.jmxremote.ssl=false
 #-J-Dwicket.configuration=development
+
+# setting the default directory for volatile data to store
+default_userdir="/var/lib/GridVisData"
+
+# Set the port of the webserver, default points to port 8080, please
+# check if there are other conflicting webserver running on this port!
+webserver_port="-J-DserverPort=8080"
+
+# The following options are used by the default, they can be overridden
+# by explicit changing the value or adding more values.
+#
+#     !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+#     !!! But! There should be a good reason after this line !!!
+#     !!! to change the following default options!           !!!
+#     !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
+
+default=""
+# use server branding
+default="$default --branding server"
+# no splash screen, no GUI window
+default="$default --nosplash --nogui"
+# the window name
+default="$default -J-Dvisualvm.display.name=GridVisService"
+# don't show exception windows
+default="$default -J-Dnetbeans.exception.report.min.level=99999 -J-Dnetbeans.exception.alert.min.level=99999"
+# set some loglevel to 'INFO'
+default="$default -J-D.level=INFO"
+default="$default -J-Dorg.netbeans.core.startup.TopLogging.level=INFO"
+default="$default -J-Dorg.netbeans.core.startup.NbEvents.level=INFO"
+# set usage of Apache Commons Logging and Log4J
+default="$default -J-Dorg.apache.commons.logging.Log=org.apache.commons.logging.impl.Jdk14Logger"
+# set loglevel for janitza modules to 'WARNING'
+default="$default -J-Dde.janitza.pasw.device.Communication.level=WARNING"
+# using AWT headless mode for the X-Server
+default="$default -J-Djava.awt.headless=true"
+# setting various storing settings
+default="$default -J-DSkipStoringEndBeforeStart"
+default="$default -J-DSkipStoringTimeInFuture"
+default="$default -J-DSkipStoringTimeTooFarInPast"
+default="$default -J-DSkipStoringTimebaseInvalid"
+
+# some Janitza specific settings
+default="$default -J-Dpasw-feature-energy-calc-builder=true -J-Dpasw-feature-ivalue-flat-persistence=true -J-Dpasw-feature-linechart-v2=true"
+default="$default -J-Dnetbeans.mainclass=de.janitza.pasw.arg.catcher.MainClass -J-Djava.util.logging.config.class=de.janitza.pasw.pasw.logback.JulHandler"
+
+# prefer IPv4 usage
+default="$default -J-Djava.net.preferIPv4Stack=true"
+
+## options for the Java memory management
+# minimum heap size memory 64MB
+default="$default -J-Xms64m"
+# maximum heap size memory 1024MB
+default="$default -J-Xmx1024m"
+
+# you may set the following options if you doing some development,
+# uncomment the lines if needed
+#default="$default -J-Dcom.sun.management.jmxremote.port=3333 -J-Dcom.sun.management.jmxremote.authenticate=false -J-Dcom.sun.management.jmxremote.ssl=false J-Dwicket.configuration=development"
 # for development purposes you may wish to append: -J-Dnetbeans.logger.console=true -J-ea
 
-# default location of JDK/JRE, can be overridden by using --jdkhome <dir> switch
-#jdkhome="jre"
+# Default location of JDK/JRE, can be overridden by using --jdkhome <dir> switch.
+# Don't change anything here, the 'jre' directory points to the current
+# installed JRE on the system!
+jdkhome="jre"
 
 # clusters' paths separated by path.separator (semicolon on Windows, colon on Unices)
 #extra_clusters=
-jdkhome="jre"
 
-        
\ No newline at end of file
+# concating the options to 'default_options' so the start script can
+# read the values
+default_options="$default $webserver_port"
